sympy/core/mod.py

```diff
@@ -30,12 +30,16 @@ def eval(cls, p, q):
         from sympy.core.singleton import S
         from sympy.core.exprtools import gcd_terms
         from sympy.polys.polytools import gcd
+        from sympy.matrices.matrices import MatrixBase
 
         def doit(p, q):
             """Try to return p % q if both are numbers or +/-p is known
             to be less than or equal q.
             """
 
+            if isinstance(p, MatrixBase):
```

Comment:
It shouldn't be necessary to import and special-case MatrixBase here. The lines 55+  below already provide a way for classes to define their own modulus evaluation: see
```
if hasattr(p, '_eval_Mod'):
    rv = getattr(p, '_eval_Mod')(q)
```